include/spdlog/details/os.h

```diff
@@ -128,6 +129,65 @@ constexpr inline unsigned short eol_size()
 }
 #endif
 
+//Return true and list if filename contains relative
+inline bool dir_check(const std::string& filename, std::list<std::string>& dirs)
+{
+#ifdef __linux__
+    std::string directory;
+    std::string::size_type index, previndex = 0, size;
+
+    index = filename.find("/", previndex);
+    if(!index)
```

Comment:
if the filename contains the absolute path, there is a "/" in front of the filename and a "/" will be appended into the list (the "/" will be passed as the argument of the 'mkdir' function).

if the directory which is the argument of the 'mkdir' function already exists, the 'errno' will be set as the 'EEXIST', So i put some code to ignore it.

As a result, the "/" will be passed as the argument of the 'chdir' function which is next to the 'mkdir' function